plugins/migration/bin/command.js

```diff
@@ -0,0 +1,111 @@
+ */
+
+import chalk from 'chalk';
+import semver from 'semver';
+import {Command} from 'commander';
+
+// TODO: Not sure the best way to access the JSON from node.
+import {createRequire} from 'module';
+const require = createRequire(import.meta.url);
+const scriptPackageJson = require('../package-lock.json');
+
+
+const SCRIPT_NAME = scriptPackageJson.name;
+
+const ROOT_COMMAND = new Command(SCRIPT_NAME)
+    .version(
+        scriptPackageJson.version,
+        '-v, --version',
+        'output the version of this script')
+    .usage(`${chalk.blue('<migration>')} ` +
+        `${chalk.green('<from-version> <to-version>')} [options] ` +
+        `${chalk.green('<file...>')}`);
+
+const HELP_COMMAND = new Command('help')
+    .argument('[version]', 'the version of Blockly to provide info about')
+    .description('display this help, or help for the specific version')
+    .action(function() {
+      if (this.args.length) {
+        showVersionHelp(this.args[0]);
+      } else {
+        ROOT_COMMAND.help();
+      }
+    });
+
+ROOT_COMMAND.addCommand(HELP_COMMAND);
+
+/**
+ * Data about all of the migrations for use in help text.
+ * @type {!Array<{range: string, name: string, description: string}>}
+ */
+const migrations = [];
+
+/**
+ * Logs information about migrations for the given version.
+ * @param {string} version The version to show migrations for.
+ */
+function showVersionHelp(version) {
+  const targetRange = `${version} - ${version}`;
+  if (!semver.validRange(targetRange)) {
+    console.log(`Invalid version: ${version}`);
+    return;
+  }
+
+  const ranges = migrations.map((migration) => migration.range);
+  if (!ranges.some((range) => semver.intersects(range, targetRange))) {
+    console.log(`No migrations found for version ${version}`);
+    return;
+  }
+
+  console.log(`Migrations for version ${version}:`);
+
+  for (const migration of migrations) {
+    if (!semver.intersects(migration.range, targetRange)) continue;
+    console.log(`  ${chalk.blue(migration.name)}, ${migration.description}`);
+  }
+}
+
+/**
+ * Creates a command with the basic format expected by the migrate script,
+ * and automatically adds it as a subcommand. It is then returned so it can be
+ * configured.
+ * @param {string} name The name of the subcommand.
+ * @param {string} targetVersionRange The range of target versions of Blockly
+ *     this command assists in migrating to.
+ * @param {string} description The description of the new subcommand.
+ * @return {!Command} The basic subcommand.
+ */
+export function createAndAddSubCommand(name, targetVersionRange, description) {
+  const migration = {
+    name: name,
+    description: description,
+    range: targetVersionRange,
+  };
+  migrations.push(migration);
+
+  const subCommand = new Command(name)
+      .description(description)
+      .argument('<from-version>', 'Blockly version to migrate from')
+      .argument('<to-version>', 'Blockly version to migrate to')
```

Comment:
I'd suggest that this should be optional, and default to the most recently released version.  I'd be ok with requiring `--from` and `--to` or the like

Reply:
`--from <from-version>` is now a required option rather than an argument. `--to <to-version>` is an optional argument with a required value. It defaults to `'latest'`. Working with `'latest'` and `'develop'` versions is now handled in a separate `versions.js` file so that it can be reused by future migrations.